cmd/root.go

```diff
@@ -48,13 +48,23 @@ var (
 		Short: "Istio Manager",
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
 			glog.V(2).Infof("Root flags: %#v", RootFlags)
+
+			if RootFlags.Kubeconfig == "" {
+				if os.Getenv("KUBECONFIG") != "" {
+					glog.V(2).Infof("Setting configuration from KUBECONFIG environment variable")
+					RootFlags.Kubeconfig = os.Getenv("KUBECONFIG")
+				}
+			}
+
 			Client, err = kube.NewClient(RootFlags.Kubeconfig, model.IstioConfig)
-			if err != nil {
+			if err != nil && RootFlags.Kubeconfig == "" {
```

Comment:
With the implementation before this change, it is possible to say `istioctl --kubeconfig ~/myconfig.yaml list route-rule`.  This is a problem if the developer really wants to keep a default ~/.kube/config and some alternatives for other clusters and makes a typo.  We don't want the behavior on a typo to be silently switching to a completely different default cluster.